### DISCUSSION TITLE
Auto disable/enable private dns

### DISCUSSION THREAD

**Discussion by CodeCracker-oss** [mullvad/mullvadvpn-app#8927] [Feature requests & Ideas]
Hello,

I am running version 2023.5 (F-Droid build) of Mullvad VPN on a pixel 6 Android 13.

I noticed that if Private DNS system feature is on, there will be a dns leak with mullvad vpn. Which means that feature needs to be disabled before running vpn.

The biggest concern is that, there is not even a warning that the feature is on, so those unaware may end up having dns leaks. At a minimum, i believe a warning should be implemented if private dns is detected as on, maybe even auto-directing the user to the settings page to disable it.

Private dns is a secure setting. It would be nice if mullvad were to be able to turn it off automatically, if its on when connecting vpn and when disconnecting turning it on if it was previously off. Though to do this one would need to give mullvad access to WRITE_SECURE_SETTINGS via adb, which is a one time process (at least until uninstalling the app).

As of now, i have to remember to turn it off, and back on accordingly which is just annoying.

Would this feature at least be considered at all?

Thank you

**Comment by faern** [comment]
I'm not sure I fully agree this is a "leak". If you have set up your phone explicitly to use a given DNS, it should probably use that DNS. For the same reason we have the "Custom DNS server" setting in the app. What do *you* think the downside is to keep using the private DNS even when the tunnel is up?

But I do understand the desire to be able to easily switch between "Using Mullvad and Mullvad DNS in the tunnel" vs "Do not use Mullvad, but send DNS encrypted to this specific server".

We are not going to add features specifically needing rooting or running ADB however. At least not now. That only caters to a very small subset of users.
